### PR TITLE
Feat/datazoom auto

### LINKED ISSUE #1416: [Feature] datazoom / scrollBar auto 模式优化

### PR DESCRIPTION
**Pull request by zamhown**
vutil: https://github.com/VisActor/VUtil/pull/124
vrender: https://github.com/VisActor/VRender/pull/669

[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #1416

### 🐞 Bugserver case id
655cadc3ba621bd56533f005
655c53e57bb49c240fa99b21

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: common/config/rush/pnpm-config.json [hunk 1]
```diff
@@ -89,9 +89,12 @@
    * PNPM documentation: https://pnpm.io/package_json#pnpmoverrides
    */
   "globalOverrides": {
-    // "@visactor/vrender-core": "~0.16.11",
-    // "@visactor/vrender-kits": "~0.16.11",
-    // "@visactor/vrender-components": "~0.16.11"
+    "@visactor/vutils": "0.16.8-alpha.0",
```
Comment (kkxxkk2019) [inline common/config/rush/pnpm-config.json:92]: vutils 和 vrender 都已经发布正式版本了，可以改回去

vutils

File: docs/package.json [hunk 2]
```diff
@@ -14,10 +14,10 @@
     "@visactor/vchart": "workspace:1.6.6",
     "@visactor/vchart-theme": "~1.6.6",
     "@visactor/vmind": "workspace:1.1.0",
-    "@visactor/vutils": "~0.16.10",
+    "@visactor/vutils": "~0.16.14",
     "@visactor/vrender": "~0.16.16",
     "@visactor/vrender-kits": "~0.16.16",
-    "@visactor/vgrammar": "~0.9.1",
+    "@visactor/vgrammar": "~0.9.0",
```
Comment (kkxxkk2019) [inline docs/package.json:20]: 0.9.1

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 3]
```diff
@@ -807,69 +846,55 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
...
+    const axis = this._relatedAxisComponent as CartesianAxis<any>;
+    const axisSpec = axis?.getSpec() as ICartesianBandAxisSpec | undefined;
+    const axisScale = axis?.getScale() as IBandLikeScale;
     const bandSizeResult = this._getAxisBandSize(axisSpec);
-    const { bandSize, maxBandSize, minBandSize } = bandSizeResult ?? {};
     let isShown = true;
-    const scale = this._stateScale as BandScale;
-    scale.range(this._isHorizontal ? [0, rect.width] : axisSpec.inverse ? [0, rect.height] : [rect.height, 0]);
-    if (isDiscrete(scale.type)) {
+    axisScale.range(this._isHorizontal ? [0, rect.width] : axisSpec.inverse ? [0, rect.height] : [rect.height, 0]);
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:858]: 水平不需要处理 inverse 吗

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 4]
```diff
@@ -807,69 +846,55 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
...
-          scale.minBandwidth(minBandSize, true);
-        }
-        scale.rescale();
+      if (bandSizeResult && (this._start || this._end)) {
+        axisScale.rangeFactor([this._start, this._end]);
       }
-      let [start, end] = scale.rangeFactor() ?? [];
-      if ((!start && !end) || !scale.isBandwidthFixed()) {
+      let [start, end] = axisScale.rangeFactor() ?? [];
+      if (isNil(start) || isNil(end)) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:872]: &&？

File: packages/vchart/src/component/data-zoom/data-zoom/interface.ts [hunk 5]
```diff
@@ -81,6 +81,14 @@ export interface IDataZoomSpec extends IDataZoomStyle, IDataFilterComponentSpec
...
    */
   brushSelect?: boolean;
+  /**
+   * 是否忽略轴上配置的固定 bandSize
+   * 如果置为 true，则 datazoom 可以任意改变轴的 bandSize。
+   * 但如果轴上配置了 bandSize 范围，则第一次渲染会保持 bandSize 在配置范围内。
+   * 该配置仅在 auto 设为 true 时生效。
+   * @since 1.7.0
+   */
+  ignoreBandSize?: boolean;
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-zoom/interface.ts:91]: 配置项文档补充下

Comment (kkxxkk2019) [reply packages/vchart/src/component/data-zoom/data-zoom/interface.ts:91]: demo 需要补充吗？

Comment (kkxxkk2019) [reply packages/vchart/src/component/data-zoom/data-zoom/interface.ts:91]: bugserver 录入下 case

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts (hunk 3 above)
Comment (zamhown) [reply packages/vchart/src/component/data-zoom/data-filter-base-component.ts]: 这个之前在横轴的逻辑确实是缺失的，我补上了

**Review by kkxxkk2019 — APPROVED** [review]

**Review by kkxxkk2019 — APPROVED** [review]

**Review by skie1997 — APPROVED** [review]

**Pull request merged**